controllers/datasciencecluster/datasciencecluster_controller.go

```diff
@@ -257,10 +257,17 @@ func (r *DataScienceClusterReconciler) Reconcile(ctx context.Context, req ctrl.R
 	}
 
 	// finalize reconciliation
+	operatorReleaseVersion, err := cluster.SetRelease(r.Client)
```

Comment:
I wonder if it would be useful to retrieve the version information before reconciling components? in case there are problems with certain components...